services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -87,30 +107,32 @@ func (rm *resourceManager) UpdateReplicaCount(
 	if respErr != nil {
 		return nil, respErr
 	}
-	return provideUpdatedResource(r, resp.ReplicationGroup)
+	return provideUpdatedResource(rd, resp.ReplicationGroup)
 }
 
 // newUpdate(ShardConfiguration)RequestPayload returns an SDK-specific struct for the HTTP request
 // payload of the Update API call for the resource
 func (rm *resourceManager) newUpdateShardConfigurationRequestPayload(
-	r *resource,
+	rd *resource, // desired
+	rl *resource, // latest
```

Comment:
actually, I was originally confused why you were passing the `resourceDescriptor` but then realized that you were just calling the "desired" resource parameter `rd`. Recommend calling these parameters `desired` and `latest` :) We use "rd" to indicate the `resourceDescriptor` struct

Reply:
Renamed variables.